packages/vuetify/src/components/VDivider/VDivider.ts

```diff
@@ -1,42 +1,58 @@
-// @ts-nocheck
-/* eslint-disable */
-
 // Styles
 import './VDivider.sass'
 
-// Types
-import { VNode } from 'vue'
+// Utilities
+import { defineComponent, h } from 'vue'
+import makeProps from '@/util/makeProps'
 
-// Mixins
-import Themeable from '../../mixins/themeable'
+// Composables
+import {
+  makeBorderProps,
+  makeDimensionProps,
+  useBorder,
+  useDimension,
+  useTheme,
+} from '@/composables'
 
-export default Themeable.extend({
-  name: 'v-divider',
+export default defineComponent({
+  name: 'VDivider',
 
   props: {
-    inset: Boolean,
-    vertical: Boolean,
+    ...makeBorderProps(),
+    ...makeDimensionProps({
+      height: undefined,
+      maxHeight: undefined,
+      maxWidth: undefined,
+      width: undefined,
+    }),
```

Comment:
What is the expected behavior when these props in this component? Is it the width/height of the actual divider line? Or is it the width/height of the component/element?

If it is the element dimension then I would expect the actual divider line to be centered correctly inside it, which it's not since it uses border. I'm not sure I understand this scenario though. If I needed to pad a divider my first instinct would be to use padding/margin classes.

If it's the divider dimension then it currently doesn't work since again it uses border to draw it. In this scenario I would probably suggest that we use props `width` and `length` instead. `width` would be the thickness of the divider line, and `length` the length of it, which works in both vertical and horizontal directions.

(also this could just be `makeDimensionProps()`)

Reply:
@jacekkarczmarczyk proposed `thickness` instead of `width` to reduce confusion